File: POH7/Megane/Main.java
// Try POH
// author: Leonardone @ NEETSDKASU
import java.io.*;
import java.util.*;
import java.math.*;
import java.text.*;

public class Main {
    public static void main(String[] args) throws Exception {
        Scanner in = new Scanner(System.in);
        
        int n = in.nextInt(); in.nextLine();
        
        String[] q = new String[n];
        
        for (int i = 0; i < n; i++) {
            q[i] = in.nextLine();
        }
        
        int m = in.nextInt(); in.nextLine();
        
        String[] p = new String[m];
        
        for (int i = 0; i < m; i++) {
            p[i] = in.nextLine();
        }
        
        outerloop:
        for (int i = 0; i <= n - m; i++) {
            innerloop:
            for (int j = 0; (j = q[i].indexOf(p[0], j)) >= 0; j++) {
                for (int k = 1; k < m; k++) {
                    int v = q[i + k].indexOf(p[k], j);
                    if (v < 0) {
                        continue outerloop;
                    }
                    if (v != j) {
                        continue innerloop;
                    }  
                }
                System.out.printf("%d %d%s", i, j / 2, System.lineSeparator());
            }
        }
        
    }
}

File: POH7/Megane/Main.js
// Try POH
// author: Leonardone @ NEETSDKASU
process.stdin.resume();
process.stdin.setEncoding('utf8');
process.stdin.on('data',function(c){main(c)});

function main(chunk) {
    var i, j, k, x;
    var s = chunk.toString().trim().split('\n');
    var n = parseInt(s[0]);
    var q = [];
    for (i = 0; i < n; i++) {
        q[i] = s[i + 1];
    }
    var m = parseInt(s[n + 1]);
    var t = {}, p = "";
    for (i = 0; i < m; i++) {
        p = s[(n + 2) + i];
        for (j = 0; j <= n - m; j++) {
            for (k = 0; (k = q[i + j].indexOf(p, k)) >= 0; k++) {
                x = j.toString() + ' ' + Math.floor(k / 2).toString();
                if (t[x] === undefined) {
                    t[x] = 0;
                }
                t[x]++;
            }
        }
    }
    for (x in t) {
        if (t[x] == m) {
            console.log(x);
        }
    }
}
